Market update 6

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * TradingView now adapts to the current theme and platform, enhancing visual consistency.
  * WebView improved for iOS 16 and earlier, ensuring correct URL handling and script injection.
  * Liquidity filter now validates input to prevent minimum values exceeding maximums, with clear error messaging.
  * Right column in Market desktop layout is now scrollable for better usability.

* **Improvements**
  * All market-related currency displays now use the user's preferred currency symbol instead of a fixed dollar sign.
  * Enhanced formatting for numeric values in holder and transaction components, reflecting user settings.
  * Refined string handling for market volume labels.
  * Improved tab rendering logic in InformationTabs for more accurate display.

* **Bug Fixes**
  * Removed unnecessary console logs for cleaner operation.
  * Fixed extra space in injected JavaScript for more precise execution.

* **Chores**
  * Minor internal refactoring and utility function additions for maintainability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
feat: Enhance InformationTabs and LiquidityFilterContent components with validation logic

- Added parameter validation in InformationTabs to return an empty array if tokenAddress or networkId is missing.
- Implemented validation logic in LiquidityFilterContent to ensure min value is not greater than max value, displaying an error message when validation fails.
- Updated apply button in LiquidityFilterContent to be disabled when validation errors are present, improving user experience.

limichange committed:
fix: Update error message in LiquidityFilterContent validation logic

- Changed error message ID to reflect that the minimum value must be greater than the specified value, improving clarity for users.

limichange committed:
feat: Add number formatting utility and update LiquidityFilterContent to use it

- Introduced a new utility function `formatNumberToKM` to format numbers into 'K' and 'M' units.
- Updated `LiquidityFilterContent` to convert min and max values to formatted strings before applying the filter, enhancing user experience.

limichange committed:
refactor: Clean up console logs and improve message handling in TradingView components

- Removed unnecessary console log statements from `TradingViewV2` and `useTradingViewV2` to streamline the code.
- Updated message handling in `InpageProviderWebView` to ensure payload is correctly formatted before passing to the receive handler, enhancing reliability.

limichange committed:
feat: Integrate theme variant support in TradingViewV2 component

- Added `useThemeVariant` hook to retrieve the current theme.
- Updated URL parameters in `TradingViewV2` to include the selected theme, enhancing customization options for users.

limichange committed:
fix: Remove unnecessary formatter options in TokenDetailHeaderRight component

- Eliminated the `formatterOptions` prop from the `NumberSizeableText` component, streamlining the code and improving clarity.

limichange committed:
fix: Remove unnecessary currency formatter options in TransactionItem components

- Eliminated the `currency` property from the `formatterOptions` in `TransactionItemNormal` and `TransactionItemSmall` components, simplifying the code.
- Updated `VolumeRow` and `PortfolioSection` components to remove redundant currency formatter options, enhancing code clarity.

limichange committed:
feat: Enhance market detail components with currency formatting support

- Integrated `useSettingsPersistAtom` to access currency settings in `HolderItem`, `TransactionItemNormal`, `VolumeRow`, and `TokenDetailHeaderRight` components.
- Updated number formatting to utilize the selected currency for displaying amounts and values, improving user experience and consistency across the market detail view.

revan-zhang commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/8338b8d5-47b7-4e4b-9765-1f0eb1d00248?refs=a39906784869593800e504990384fe81798d407c&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/8338b8d5-47b7-4e4b-9765-1f0eb1d00248/license?refs=a39906784869593800e504990384fe81798d407c&source=prChecksComment)

limichange committed:
refactor: Update WebView components and enhance logging capabilities

- Removed unnecessary glog privacy bundle references from the Xcode project.
- Improved the `WebView` component by adding a reference for better script injection handling.
- Introduced a new utility function for console logging in the WebView, allowing dynamic logging of messages with different severity levels.
- Updated the `DesktopLayout` to utilize `ScrollView` for better layout management, enhancing user experience in the market detail view.

limichange committed:
refactor: Simplify layout structure in DesktopLayout component

- Removed unnecessary ScrollView wrapping around the TradingView and Info tabs, improving layout efficiency.
- Adjusted the height of the Info tabs to use a percentage-based value for better responsiveness.

limichange committed:
refactor: Simplify WebView component by removing unused imports and streamlining load handling

- Removed unused `WebViewNavigationEvent` import to clean up the code.
- Consolidated the load handling logic by eliminating redundant setTimeout calls, improving readability and performance.

limichange committed:
Merge branch 'x' into market-update-6

originalix commented on packages/kit/src/views/Market/MarketHomeV2/utils.ts:
```diff
@@ -18,6 +18,23 @@ export const parseValueToNumber = (value: string): number => {
+    return thousands % 1 === 0 ? `${thousands}K` : `${thousands.toFixed(1)}K`;
+  }
+  return value.toString();
+};
```
numberUtils.formatMarketCap

limichange replied:
这个是在流动性过滤器里的一个值，不是单纯显示，跟 formatMarketCap 的格式化有点区别。不过我试试改一下。

limichange replied:
去掉了。

limichange committed:
chore: Update Podfile.lock and Xcode project for glog privacy bundle integration

- Updated the checksum for the DoubleConversion and glog dependencies in Podfile.lock.
- Added glog privacy bundle references in the Xcode project to ensure proper resource handling.

limichange committed:
fix: Adjust timeout duration in WebView component for improved load timing

- Reduced the timeout duration from 550ms to 500ms to optimize the script injection timing in the WebView component.

limichange committed:
refactor: Reorder utility imports in LiquidityFilterContent component for consistency

limichange committed:
feat: Enhance TransactionItemSmall component to utilize currency settings

- Integrated currency settings from the global state into the TransactionItemSmall component.
- Updated formatter options for transaction amounts and prices to reflect the selected currency, improving user experience in transaction history display.

limichange committed:
refactor: Update number formatting in LiquidityFilterContent and clean up utility functions

- Removed the formatNumberToKM utility function to streamline number formatting.
- Integrated number formatting directly in the LiquidityFilterContent component using the numberFormat utility.
- Enhanced WebView component to ensure proper type checking for Platform.Version.

originalix committed:
Merge branch 'x' into market-update-6

limichange committed:
refactor: Clean up WebView component by removing console log injection method

- Removed the `injectConsoleLog` method from the `IWebViewRef` type and the corresponding implementation in the `NativeWebView` component.
- Reorganized import statements for better readability.

limichange committed:
refactor: Remove unused console log injection method from WebView utilities

- Eliminated the `createConsoleLogInjectedScript` function from the WebView utilities, streamlining the codebase.
- Updated import statements in the `NativeWebView` component to reflect the removal of the unused method.
